File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        p = 1
        z = 0
        for i in nums:
            if(i == 0):
                z = z+1
            else:
                p = p*i
        res = []
        if z == 0:
            for i in nums:
                res.append(p//i)
        else:
            for i in nums:
                if i == 0 and z == 1:
                    res.append(p)
                else:
                    res.append(0)

        return res 

File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */



class Solution {
public:
    void depth_counter(TreeNode* node, int pres, int* res){
        if(node == NULL){
            *res = max(pres,*res);
            return;
        }
        depth_counter(node->left, pres+1, res);
        depth_counter(node->right, pres+1, res);
    }
    int maxDepth(TreeNode* root) {
        int res = 0;
        int pres = 0;
        depth_counter(root, pres, &res);
        return res;
    }
};
